### PR TITLE
Bilateral filter on disparity image

### PR DESCRIPTION
**Pull request by tanismar**
Implemented the option of processing the output disparity image with a fast edge-preserving bilateral filter in order to reduce noise (based on http://people.csail.mit.edu/sparis/bf/). Filter parameters as well as activation of the filter can be controlled by rpc, with added commands bilatfilt (sigmaColor sigmaSpace) and doBLF (bool), respectively. Speed is reduced for small values of the parameters, but for default and seemingly more useful ones (around 10.0 10.0) speed reduction is minimal.

### CODE REVIEW

File: modules/SFM/SFM.cpp [hunk 1]
```diff
@@ -411,9 +417,19 @@ bool SFM::updateModule()
             
     if (outDisp.getOutputCount()>0)
     {
-        outputD=stereo->getDisparity();
+        outputD = stereo->getDisparity();
         ImageOf<PixelMono> &outim=outDisp.prepare();
-        outim.wrapIplImage(&outputD);
+        if (doBLF)
+        {
+            IplImage* outputDpt = &outputD;
```
Comment (vtikha) [inline modules/SFM/SFM.cpp]: @tanismar don't you need a `cvReleaseImage(&outputDpt);` when you have finished with this?

File: app/scripts/SFM_APP.xml.template [hunk 2]
```diff
@@ -1,6 +1,6 @@
```
Comment (vtikha) [inline app/scripts/SFM_APP.xml.template]: @tanismar I would not add the SFM_APP.xml.template to the repo. This should be the same original SFM template with the flag for the bilateral filter on or off.

**Comment by tanismar** [comment]
Checked... cvReleaseImage is not needed: The code is not leaking memory, and if added, it crashes. I think for images not created as IplImage but as is the case, from Mat, Mat takes care of releasing memory and cleaning up. Name of template set back to SFM. Also did a minor change to have the filter ON by default without ini file, as there is not SFM.ini.

**Comment by vtikha** [comment]
Ok perfect, by looking at it quickly it seemed it needed it. I know Mat do not need it but since it was an IplImage i thought it might.

**Comment by pattacini** [comment]
The SFM.xml.template has been removed completely!

**Comment by pattacini** [comment]
I've added it up again.

**Comment by pattacini** [comment]
@robotology/wysiwyd-developers be aware of this merged PR introducing the Bilateral filter for smoothing out the resulting disparity.

**Comment by Tobias-Fischer** [comment]
Thanks @pattacini! I guess we don't specifically have to modify anything in the `iol2opc` pipeline, do we?

**Comment by pattacini** [comment]
We don't, indeed.

**Pull request merged**